docs/root/source/private-data.rst

```diff
@@ -0,0 +1,100 @@
+- Encrypted data can’t be indexed or searched by MongoDB. (It can index and search the ciphertext, but that’s not very useful.) One might use homomorphic encryption to index and search encrypted data, but MongoDB doesn’t have any plans to support that any time soon.
+
+System Example 1
+~~~~~~~~~~~~~~~~
+
+Encrypt the data with a symmetric key and store the ciphertext on-chain (in ``metadata`` or ``asset.data``). To communicate the key to a third party, use their public key to encrypt the symmetric key and send them that. They can decrypt the symmetric key with their private key, and then use that symmetric key to decrypt the on-chain ciphertext.
+
+The reason for using a symmetric key along with public/private keypairs is so the ciphertext only has to be stored once.
+
+System Example 2
+~~~~~~~~~~~~~~~~
+
+This example uses `proxy re-encryption <https://en.wikipedia.org/wiki/Proxy_re-encryption>`_:
+
+#. MegaCorp encrypts some data using its own public key, then stores that encrypted data (ciphertext 1) in a BigchainDB network.
+#. MegaCorp wants to let others read that encrypted data, but without ever sharing their private key and without having to re-encrypt themselves for every new recipient. Instead, they find a “proxy” named Moxie, to provide proxy re-encryption services.
+#. Zorban contacts MegaCorp and asks for permission to read the data.
+#. MegaCorp asks Zorban for his public key.
+#. MegaCorp generates a “re-encryption key” and sends it to their proxy, Moxie.
+#. Moxie (the proxy) uses the re-encryption key to encrypt ciphertext 1, creating ciphertext 2.
+#. Moxie sends ciphertext 2 to Zorban (or to MegaCorp who forwards it to Zorban).
+#. Zorban uses his private key to decrypt ciphertext 2, getting the original un-encrypted data.
+
+Note:
+
+- The proxy only ever sees ciphertext. They never see any un-encrypted data.
+- Zorban never got the ability to decrypt ciphertext 1, i.e. the on-chain data.
+- There are variations on the above flow.
+
+System Example 3
+~~~~~~~~~~~~~~~~
+
+This example uses `erasure coding <https://en.wikipedia.org/wiki/Erasure_code>`_:
+
+#. Erasure-code the data into n pieces.
+#. Encrypt each of the n pieces with a different encryption key.
+#. Store the n encrypted pieces on-chain, e.g. in n separate transactions.
+#. Share each of the the n decryption keys with a different party.
+
+If k < N of the key-holders gets and decrypts k of the pieces, they can reconstruct the original plaintext. Less than k would not be enough.
+
+System Example 4
+~~~~~~~~~~~~~~~~
+
+This setup could be used in an enterprise blockchain scenario where a special node should be able to see parts of the data, but the others should not.
+
+- The special node generates an X25519 keypair (or similar asymmetric *encryption* keypair).
+- A BigchainDB end user finds out the X25519 public key (encryption key) of the special node.
```

Comment:
`finds out` can be `queries`. just to make it clear that the encryption public key of the node will be published by the node.

Reply:
I wanted to leave it vague (how the end user finds out the key). Maybe they get it from the node via SMS, Slack or fax.